libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.helpers.ts

```diff
@@ -0,0 +1,182 @@
+  }
+  const rawAmount = massIdTotalValue
+    .multipliedBy(creditsDocumentUnitPrice)
+    .minus(participantsAmount);
+  const rawPercentage = new BigNumber(100).minus(participantsPercentage);
+
+  return {
+    amount: formatDecimalPlaces(BigNumber.maximum(rawAmount, 0)),
+    percentage: formatDecimalPlaces(BigNumber.maximum(rawPercentage, 0)),
+  };
+};
+
+export const calculateAmount = (options: {
+  creditsDocumentUnitPrice: NonZeroPositive;
+  massIdCertificateValue: BigNumber;
+  massIdPercentage: string;
+  participantAmount: string | undefined;
+}): NonEmptyString => {
+  const amount = formatDecimalPlaces(
+    formatPercentage(new BigNumber(options.massIdPercentage))
+      .multipliedBy(options.massIdCertificateValue)
+      .multipliedBy(new BigNumber(options.creditsDocumentUnitPrice)),
+  );
+
+  return amount.plus(options.participantAmount ?? 0).toString();
+};
+
+export const addParticipantRemainder = (options: {
+  actors: ActorsByActorType;
+  remainder: Remainder;
+}): void => {
+  const { actors, remainder } = options;
+
+  for (const [key, actor] of actors) {
+    if (actor.actorType === RewardsDistributionActorType.NETWORK) {
+      actors.set(key, {
+        ...actor,
+        amount: remainder.amount.plus(actor.amount).toString(),
+        percentage: remainder.percentage.plus(actor.percentage).toString(),
+      });
+
+      break;
+    }
+  }
+};
+
+const calculateCertificateTotalValue = (certificateValues: BigNumber[]) =>
+  sumBigNumbers(certificateValues);
+
+export const aggregateMassIdCertificatesRewards = (
+  creditsDocumentUnitPrice: NonZeroPositive,
+  resultContentsWithMassIdCertificateValue: ResultContentsWithMassIdCertificateValue[],
+): AggregateMassIdRewards => {
+  const actors: ActorsByActorType = new Map();
+
+  const certificateTotalValue = calculateCertificateTotalValue(
+    resultContentsWithMassIdCertificateValue.map(
+      ({ massIdCertificateValue }) => massIdCertificateValue,
+    ),
+  );
+
+  const creditTotal = formatDecimalPlaces(
+    certificateTotalValue.multipliedBy(creditsDocumentUnitPrice),
+  );
+
+  for (const {
+    massIdCertificateValue,
+    resultContent: { massIdRewards },
+  } of resultContentsWithMassIdCertificateValue) {
+    for (const massIdReward of massIdRewards) {
+      const { actorType, massIdPercentage, participant } = massIdReward;
+      const actor = actors.get(`${actorType}-${participant.id}`);
+
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue,
+        massIdPercentage,
+        participantAmount: actor?.amount,
+      });
+
+      actors.set(`${actorType}-${participant.id}`, {
+        actorType,
+        amount,
+        participant,
+        percentage: calculateCreditPercentage(
+          new BigNumber(amount),
+          creditTotal,
+        ),
+      });
+    }
+  }
+
+  return {
+    actors,
+    massIdTotalValue: certificateTotalValue,
```

Comment:
I think that it is better to use `certificateTotalValue` as the attribute name instead of `massIdTotalValue` to avoid inconsistencies.